analytics/src/test/java/com/segment/analytics/integrations/BasePayloadTest.kt

```diff
@@ -0,0 +1,320 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2014 Segment.io, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.segment.analytics.integrations
+
+import androidx.annotation.Nullable
+import com.google.common.collect.ImmutableList
+import com.google.common.collect.ImmutableMap
+import com.nhaarman.mockitokotlin2.any
+import com.segment.analytics.integrations.BasePayload.Type
+import java.util.Date
+import org.assertj.core.api.Assertions
+import org.assertj.core.data.MapEntry
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+
+class BasePayloadTest {
+
+    private lateinit var builders: List<BasePayload.Builder<out BasePayload, out BasePayload.Builder<*, *>>>
+
+    @Before
+    fun setUp() {
+        builders =
+            ImmutableList.of(
+                AliasPayload.Builder().previousId("previousId").userId("userId"),
+                TrackPayload.Builder().event("event"),
+                ScreenPayload.Builder().name("name"),
+                GroupPayload.Builder().groupId("groupId"),
+                IdentifyPayload.Builder().traits(ImmutableMap.of<String, Any>("foo", "bar"))
```

Comment:
there is a better kotlin function for creating immutable maps ->  [`mapOf`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/map-of.html), lets use that